trl/trainer/iterative_sft_trainer.py

```diff
@@ -0,0 +1,393 @@
+        config: IterativeSFTConfig = None,
+        model: PreTrainedModel = None,
+        tokenizer: PreTrainedTokenizerBase = None,
+        optimizer: Optional[torch.optim.Optimizer] = None,
+        data_collator: Optional[DataCollator] = None,
+        lr_scheduler: Optional[torch.optim.lr_scheduler._LRScheduler] = None,
+    ):
+        """
+        Initialize IterativeSFTTrainer.
+
+        Args:
+            config (`IterativeConfig`):
+                Configuration object for IterativeTrainer.
+            model (`PreTrainedModel`):
+                Hugging Face transformer model.
+            tokenizer (`transformers.PreTrainedTokenizerBase`):
+                Hugging Face tokenizer.
+            optimizer (Optional[`torch.optim.Optimizer`]):
+                Optimizer used for training. If `None`, `Adam` is used as default.
+            data_collator (Optional['DataCollator']):
+                Data collator function.
+            lr_scheduler (Optional[`torch.optim.lr_scheduler`]):
+                Learning rate scheduler used for training.
+        """
+
+        self.config = config
+
+        # initial seed for reproducible experiments
+        set_seed(config.seed)
+
+        # Step 0: check positional arguments validity
+        if not isinstance(config, IterativeSFTConfig):
+            raise ValueError(f"config must be a IterativeConfig, got {type(config)}")
+        if not isinstance(tokenizer, (PreTrainedTokenizerBase)):
+            raise ValueError(
+                f"tokenizer must be a PreTrainedTokenizerBase like a PreTrainedTokenizer or a PreTrainedTokenizerFast, got {type(tokenizer)}"
+            )
+        if not isinstance(model, PreTrainedModel):
+            raise ValueError(f"model must be a PreTrainedModel, got {type(model)}")
+        if not model.can_generate():
+            warnings.warn(
+                f"The current model class {type(model)} is not compatible with `.generate()`"
+                "Please make sure that this is intended."
+            )
+
+        # Step 1: Initialize Accelerator
+        self.accelerator = Accelerator(
+            log_with=config.log_with,
+            gradient_accumulation_steps=config.gradient_accumulation_steps,
+            project_config=ProjectConfiguration(**config.project_kwargs),
+            **config.accelerator_kwargs,
+        )
+
+        is_using_tensorboard = config.log_with is not None and config.log_with == "tensorboard"
+
+        self.accelerator.init_trackers(
+            config.tracker_project_name,
+            config=dict(trl_ppo_trainer_config=config.to_dict()) if not is_using_tensorboard else config.to_dict(),
+            init_kwargs=config.tracker_kwargs,
+        )
+
+        self.model = model
+        self.model_params = filter(lambda p: p.requires_grad, self.model.parameters())
+        self.is_encoder_decoder = hasattr(self.model, "is_encoder_decoder")
+        self.is_peft_model = getattr(self.model, "is_peft_model", False)
```

Comment:
```suggestion
        self.is_peft_model = is_peft_available() and isinstance(self.model, PeftModel)
```
And make sure to import peftModel if peft is available on the top of the file

Reply:
Indeed, thanks